src/hooks/usePokemonList.ts

```diff
@@ -0,0 +1,7 @@
+import { useQuery } from "react-query";
+import { PokemonItem } from "../interfaces/pokemonItem";
+import { getPokemonList } from "../services/pokemonListService";
+
+export default function usePokemonList() {
+  return useQuery<PokemonItem[], Error>("pokemon-list", getPokemonList);
```

Comment:
In teoria non dovrebbe esserci bisogno di specificare il tipo di ritorno qui, dato che `getPokemonList` ha gia' una sua firma ben definita. In teoria TS dovrebbe essere abbastanza smart da dedurlo da solo ;)

Reply:
Ho specificato i generics in quanto error restituito da useQuery mi veniva segnalato di tipo Unknow. Posso togliere e aggiungere if (error instanceof Error) prima di richiamare error.message?